### ISSUE TITLE
dsp: `assetId` is null on new transfer request

### ISSUE THREAD

**Issue by ndr-brt** [eclipse-edc/Connector#2959]
# Bug Report

## Describe the Bug
When a new transfer is requested, the `assetId` set in the `DataRequest` is null, because that information is not set in the `TransferRequestMessage`.

See `TransferProcessProtocolServiceImpl`

### Expected Behavior
The `TransferProcessProtocolServiceImpl.requestedAction` should get the agreement from the `contractId` and extract the `assetId` from there.

**Comment by ndr-brt** [comment]
EDIT: maybe it could be extracted from the `contractId`, that now contains the `assetId` as well

**Comment by jimmarino** [comment]
I think this information has to be set. We can't assume the asset id is in the EDC format since the request could be sent from a different runtime type.

**Comment by jimmarino** [comment]
Yep, previously, I thought this was about the offerId but contracted we can do this.
